MGMT-8156: Extract strings for localization (#1399)
openshift-assisted/assisted-ui-lib: I18N.md

## i18n

Internationalization is implemented with [i18next](https://www.i18next.com/) and

> [react-i18next](https://react.i18next.com/). The i18n namespace must be declared in
> TRANSLATION_NAMESPACE property to avoid naming conflicts.

Reviewer: ```suggestion
[react-i18next](https://react.i18next.com/). The i18n namespace must be declared at build-time in
an environment variable called: `TRANSLATION_NAMESPACE` in order to avoid naming conflicts.
```

Author: changes done

- Environment variables:

  ```
    TRANSLATION_NAMESPACE: required, name of translation namespace
  ```

> - We create the wrapper src/common/i18n/cim-18n that use the `useTranslation` hook with our

Author: ok!

  namespace.

> - Use in our classes :

Reviewer: ```suggestion
- Use in our components:
```

Author: changes done

## i18n

> See [i18n](link) for information on our internationalization tools and guidelines

Reviewer 2: ```suggestion
```

> We use [react-i18next](https://react.i18next.com/) to translate messages in assisted-ui-lib. The

Reviewer 2: ```suggestion
Internationalization is implemented with [i18next](https://www.i18next.com/) and [react-i18next](https://react.i18next.com/). The
```

> i18n namespace must be declared in TRANSLATION_NAMESPACE property to avoid naming conflicts.

Reviewer 2: We need to verify https://github.com/openshift-assisted/assisted-ui-lib/pull/1399/files#r883540666 , maybe we can provide some meaningful default for TRANSLATION_NAMESPACE  ?

- Environment variables:

```

>   TRANSLATION_NAMESPACE: required, name of translation namespace

Reviewer 2: We need to verify https://github.com/openshift-assisted/assisted-ui-lib/pull/1399/files#r883540666 , maybe we can provide some meaningful default for TRANSLATION_NAMESPACE ?

```

To test to standalone we need to put TRANSLATION_NAMESPACE=cim_namespace

- We create the wrapper src/common/i18n/cim-18n that use the `useTranslation` hook with our
  namespace as follows:

```tsx
import { useTranslation as useReactI18NextTranslation } from 'react-i18next';

export function useTranslation() {
  return useReactI18NextTranslation(`${process.env.TRANSLATION_NAMESPACE}`);
}

> ```

Reviewer 2: I think we dont want to copy-paste the implementation of our `useTranslation` hook as it can change often. Better would be to add a link here

- Use in our classes :

```tsx
import { useTranslation } from '../../i18n/cim-i18n';
....
 <Button key="submit" onClick={submitForm}>
  {t('ai:Generate Discovery ISO')}
</Button>
```

- File scripts/i18n-parser.config.js allow to generate the translation files automatically inside
  root/locales folder:

```js
module.exports = {
  contextSeparator: '_',
  createOldCatalogs: false,
  keySeparator: false,
  locales: ['en'],
  defaultNS: 'cim_namespace', // the default file for strings when using useTranslation, etc
  namespaceSeparator: '~',
  reactNamespace: false,
  useKeysAsDefaultValue: true,
  output: './locales/$LOCALE/translation.json',
  sort: true,
};

> ```

Reviewer 2: I dont think we need to have this part in the readme. Anybody can just look at `yarn i18n` script (which is mentioned below) and see what is going on .

Author: new proposal of i18n.md:
![image](https://user-images.githubusercontent.com/11390125/171004061-1fd0c95f-baeb-4719-8787-0fd57eeb180b.png)
